sqlglot/expressions.py

```diff
@@ -5373,9 +5373,14 @@ class Upper(Func):
     _sql_names = ["UPPER", "UCASE"]
 
 
+# https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/VARIANCE.html
+# oracle variance returns 0 if called on 1-element input set
+# var_samp returns NULL if called 1-element input set
 class Variance(AggFunc):
     _sql_names = ["VARIANCE", "VARIANCE_SAMP", "VAR_SAMP"]
 
+    arg_types = {"this": True, "return_zero": False}
+
```

Comment:
Is the pattern here to define a `variance` rule in all the other dialects to ignore this argument?

Reply:
You would need to match Oracle's semantics so that the transpilation is correct when you're going from Oracle -> X. So, produce something like `IFNULL(VAR_SAMP(x), 0)`. I think we need to override `variance_sql` in the base class and control its behavior with class constants to do this properly.